### PR TITLE
test: add e2e test to verify empty outputRelativeDirectories lead to no output

### PR DESCRIPTION
**Pull request by godobyte**
### What was the problem/requirement? (What/Why)
ouputRelativeDirectories is an optional field in [ManifestProperties](https://docs.aws.amazon.com/deadline-cloud/latest/APIReference/API_ManifestProperties.html) as part of CreateJob API model. If not provided upon calling CreateJob, the value will be None. When the list is None, nothing gets recognize as upload for upload.

This can be easily overlooked, we should have a test to guard the expected behaviour.

### What was the solution? (How)
Add tests to guard the expected behaviour when `ouputRelativeDirectories` is None.

### What is the impact of this change?
Better test coverage.

#### Verified both windows and linux tests pass.
```
test/e2e/test_job_submissions.py::TestJobSubmission::test_job_attachments_no_output_relative_directories
=================== 1 passed, 1 warning in 342.44s (0:05:42) ===================
```
#### Verified no job output

windows
```
 ~> deadline job download-output --job-id job-1bd1c08da4f94ec3a991d8e6c6fc9572
Downloading output from Job 'empty-outputRelativeDirectories-test-windows'
There are no output files available for download at this moment. Please verify that the Job/Step/Task you are trying to download output from has completed successfully.
```
linux
```
 ~> deadline job download-output --job-id job-c1fc3dee3df54c6f9f60a90cb7a00272
Downloading output from Job 'empty-outputRelativeDirectories-test-linux'
There are no output files available for download at this moment. Please verify that the Job/Step/Task you are trying to download output from has completed successfully.
```

### Is this a breaking change?

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/e2e/test_job_submissions.py [hunk 1]
```diff
@@ -3406,3 +3406,156 @@ def test_worker_create_job_API_call_windows(
...
+        # Upload manifests
+        s3.put_object(
+            Bucket=s3_bucket,
+            Key=f"{s3_prefix}/Manifests/manifest1hash",
+            Body=json.dumps(manifest),
+            ExpectedBucketOwner=test_runner_identity["Account"],
+        )
+
+        # Create and submit job using boto3 directly
+        template = {
```
Comment (leongdl) [inline test/e2e/test_job_submissions.py:3467]: Is there anyway to put this job template to a file and just source it in code?

Comment (leongdl) [reply test/e2e/test_job_submissions.py:3467]: This would make a good job bundle to submit.

Comment (godobyte) [reply test/e2e/test_job_submissions.py:3467]: We could, but I'd track that task separately because this test currently uses CreateJob (instead of bundle submit) to also test any s3 input path, which is not applicable with `bundle submit`.

**Review by AWS-Samuel — APPROVED** [review]

File: test/e2e/test_job_submissions.py (hunk 1 above)
Comment (leongdl) [reply test/e2e/test_job_submissions.py:3467]: ok no problem.

**Review by leongdl — APPROVED** [review]
